packages/devextreme/js/data/utils.js

```diff
@@ -232,6 +232,15 @@ const isGroupOperator = function(value) {
     return value === 'and' || value === 'or';
 };
 
+export const isUniformEqualsByOr = function(crit) {
```

Comment:
I feel like, ideally, we need separate unit tests for this method to make sure it rejects all the non-uniform criteria. It's the most critical point here where we can avoid (or not avoid) BCs.

Like for cases with (1) different properties (2) different operands, other than or (3) different criteria, other than = (4) more depth

Reply:
ok.. i will add test for it